include/range/v3/utility/iterator.hpp

```diff
@@ -974,10 +955,11 @@ namespace ranges
                 {
                     return that.it_ - it_;
                 }
-                CONCEPT_REQUIRES(Readable<I>())
+                template<typename II = I,
+                    CONCEPT_REQUIRES_(Same<I, II>() && Readable<II>())>
```

Comment:
Without this change view.set_difference and view.set_symmetric_difference hit hard errors trying to instantiate `move_into_iterator`s:
```
/usr/bin/cmake -H/home/casey/dev/range-v3 -B/home/casey/dev/range-v3/build-c3.8 --check-build-system CMakeFiles/Makefile.cmake 0
make -f CMakeFiles/Makefile2 view.set_difference
make[1]: Entering directory '/home/casey/dev/range-v3/build-c3.8'
/usr/bin/cmake -H/home/casey/dev/range-v3 -B/home/casey/dev/range-v3/build-c3.8 --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/casey/dev/range-v3/build-c3.8/CMakeFiles 1
make -f CMakeFiles/Makefile2 test/view/CMakeFiles/view.set_difference.dir/all
make[2]: Entering directory '/home/casey/dev/range-v3/build-c3.8'
make -f test/view/CMakeFiles/view.set_difference.dir/build.make test/view/CMakeFiles/view.set_difference.dir/depend
make[3]: Entering directory '/home/casey/dev/range-v3/build-c3.8'
cd /home/casey/dev/range-v3/build-c3.8 && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/casey/dev/range-v3 /home/casey/dev/range-v3/test/view /home/casey/dev/range-v3/build-c3.8 /home/casey/dev/range-v3/build-c3.8/test/view /home/casey/dev/range-v3/build-c3.8/test/view/CMakeFiles/view.set_difference.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/home/casey/dev/range-v3/build-c3.8'
make -f test/view/CMakeFiles/view.set_difference.dir/build.make test/view/CMakeFiles/view.set_difference.dir/build
make[3]: Entering directory '/home/casey/dev/range-v3/build-c3.8'
[  0%] Building CXX object test/view/CMakeFiles/view.set_difference.dir/set_difference.cpp.o
cd /home/casey/dev/range-v3/build-c3.8/test/view && /usr/bin/clang++-3.8    -I/home/casey/dev/range-v3/include  -stdlib=libc++ -std=c++14 -ftemplate-backtrace-limit=0 -Weverything -Werror -pedantic-errors -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-weak-vtables -Wno-padded -Wno-undef -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-shadow -Wno-old-style-cast -Wno-documentation -Wno-documentation-unknown-command -march=native -mtune=native   -o CMakeFiles/view.set_difference.dir/set_difference.cpp.o -c /home/casey/dev/range-v3/test/view/set_difference.cpp
In file included from /home/casey/dev/range-v3/test/view/set_difference.cpp:15:
In file included from /home/casey/dev/range-v3/include/range/v3/core.hpp:17:
In file included from /home/casey/dev/range-v3/include/range/v3/begin_end.hpp:24:
In file included from /home/casey/dev/range-v3/include/range/v3/utility/iterator.hpp:25:
In file included from /home/casey/dev/range-v3/include/range/v3/utility/iterator_traits.hpp:20:
/home/casey/dev/range-v3/include/range/v3/utility/iterator_concepts.hpp:154:53: error: no matching function for call to object of type 'const ranges::v3::adl_move_detail::indirect_move_fn'
                using rvalue_reference_t = decltype(indirect_move(std::declval<I &>()));
                                                    ^~~~~~~~~~~~~
/home/casey/dev/range-v3/include/range/v3/utility/iterator_traits.hpp:37:9: note: in instantiation of template type alias 'rvalue_reference_t' requested here
        using iterator_rvalue_reference_t = concepts::Readable::rvalue_reference_t<I>;
        ^
/home/casey/dev/range-v3/include/range/v3/utility/iterator.hpp:961:17: note: in instantiation of template type alias 'iterator_rvalue_reference_t' requested here
                iterator_rvalue_reference_t<I> move() const
                ^
/home/casey/dev/range-v3/include/range/v3/range_access.hpp:49:20: note: in instantiation of template class 'ranges::v3::detail::move_into_cursor<ranges::v3::basic_iterator<ranges::v3::detail::back_insert_cursor<std::__1::vector<MoveOnlyString, std::__1::allocator<MoveOnlyString> > > > >' requested here
            static meta::id<typename T::mixin> mixin_base_2_(int);
                   ^
/home/casey/dev/range-v3/include/range/v3/range_access.hpp:53:26: note: while substituting explicitly-specified template arguments into function template 'mixin_base_2_' 
              : decltype(range_access::mixin_base_2_<Cur>(42))
                         ^
/home/casey/dev/range-v3/include/meta/meta.hpp:140:9: note: in instantiation of template class 'ranges::v3::range_access::mixin_base_<ranges::v3::detail::move_into_cursor<ranges::v3::basic_iterator<ranges::v3::detail::back_insert_cursor<std::__1::vector<MoveOnlyString, std::__1::allocator<MoveOnlyString> > > > > >' requested here
        using _t = typename T::type;
        ^
/home/casey/dev/range-v3/include/range/v3/range_access.hpp:61:13: note: in instantiation of template type alias '_t' requested here
            using mixin_base_t = meta::_t<mixin_base_<Cur>>;
            ^
/home/casey/dev/range-v3/include/range/v3/utility/basic_iterator.hpp:309:27: note: in instantiation of template type alias 'mixin_base_t' requested here
          : range_access::mixin_base_t<Cur>
                          ^
/home/casey/dev/range-v3/include/range/v3/utility/iterator.hpp:984:45: note: in instantiation of template class 'ranges::v3::basic_iterator<ranges::v3::detail::move_into_cursor<ranges::v3::basic_iterator<ranges::v3::detail::back_insert_cursor<std::__1::vector<MoveOnlyString, std::__1::allocator<MoveOnlyString> > > > > >' requested here
            constexpr move_into_iterator<I> operator()(I it) const
                                            ^
/home/casey/dev/range-v3/test/view/set_difference.cpp:247:33: note: in instantiation of function template specialization 'ranges::v3::move_into_fn::operator()<ranges::v3::basic_iterator<ranges::v3::detail::back_insert_cursor<std::__1::vector<MoveOnlyString, std::__1::allocator<MoveOnlyString> > > > >' requested here
                       move_into(back_inserter(expected_greedy)),
                                ^
/home/casey/dev/range-v3/include/range/v3/utility/move.hpp:81:22: note: candidate template ignored: substitution failure [with I = ranges::v3::basic_iterator<ranges::v3::detail::back_insert_cursor<std::__1::vector<MoveOnlyString, std::__1::allocator<MoveOnlyString> > > >]: no matching function for call to 'indirect_move'
                auto operator()(I const &i) const
                     ^
1 error generated.
test/view/CMakeFiles/view.set_difference.dir/build.make:62: recipe for target 'test/view/CMakeFiles/view.set_difference.dir/set_difference.cpp.o' failed
make[3]: *** [test/view/CMakeFiles/view.set_difference.dir/set_difference.cpp.o] Error 1
make[3]: Leaving directory '/home/casey/dev/range-v3/build-c3.8'
CMakeFiles/Makefile2:7465: recipe for target 'test/view/CMakeFiles/view.set_difference.dir/all' failed
make[2]: *** [test/view/CMakeFiles/view.set_difference.dir/all] Error 2
make[2]: Leaving directory '/home/casey/dev/range-v3/build-c3.8'
CMakeFiles/Makefile2:7477: recipe for target 'test/view/CMakeFiles/view.set_difference.dir/rule' failed
make[1]: *** [test/view/CMakeFiles/view.set_difference.dir/rule] Error 2
make[1]: Leaving directory '/home/casey/dev/range-v3/build-c3.8'
Makefile:2643: recipe for target 'view.set_difference' failed
make: *** [view.set_difference] Error 2
```